src/common/definition.ts

```diff
@@ -64,6 +66,75 @@ export async function parseDefinition({
     }
   }
 
+  if (excludeExt) {
+    const removeSpecifiedExtensions = (obj, parent = null, parentKey: string = '') => {
+      if (typeof obj !== 'object' || obj === null) return;
+
+      for (const key in obj) {
+        if (excludeExt == key && parent) {
+          // Remove the entire operation (e.g., get, post) if specified extension is found
+          delete parent[parentKey];
+          break; // Exit the loop as the entire operation has been removed
+        } else if (typeof obj[key] === 'object') {
+          removeSpecifiedExtensions(obj[key], obj, key);
+        }
+      }
+    };
+
+    // Start the traversal from the root of the document
+    removeSpecifiedExtensions(document);
+    // Remove empty paths
+    Object.keys(document.paths).forEach(path => {
```

Comment:
would be better structured if we have this remove empty paths operation as a function and called it here.